scripts/src/owners/redhat_metadata.py

```diff
@@ -0,0 +1,89 @@
+from argparse import ArgumentParser
+from . import owners_file
+
+REQUIRED_VENDOR_LABEL = "redhat"
+REQUIRED_VENDOR_NAME = "Red Hat"
+REQUIRED_CHART_PREFIX = "redhat-"
+
+
+class RedHatOwnersFileInvalidContentsError(Exception):
+    message = "The OWNERS does not contain criteria required for Red Hatters"
+
+    def __init__(self, message):
+        if len(message) > 0:
+            self.message = message
+
+        super().__init__(self.message)
+
+
+def assert_redhat_metadata(contents):
+    """Ensures contents of owners file for Red Hat submissions
+
+    Red Hat submissions are expected to have certain metadata. - The owner label
+    must be "redhat" - The owner organizaton name must be "Red Hat" - The
+    chart's name must be prefixed with "redhat-"
+
+    This raises an exception if the metadata is incorrect.
+
+    Args:
+        contents: The contents of the owners
+          file to be evaluated. This should have already
+          been read from disk and yaml.loaded
+
+    Raises:
+        RedHatOwnersFileInvalidContentsError: In cases where the
+          content is not as expected.
+    Returns:
+        Nothing.
+    """
+
+    if not owners_file.get_chart(contents).startswith(REQUIRED_CHART_PREFIX):
+        raise RedHatOwnersFileInvalidContentsError(
+            f"The chart name must be prefixed with '{REQUIRED_CHART_PREFIX}'."
+        )
+
+    if owners_file.get_vendor_label(contents) != REQUIRED_VENDOR_LABEL:
+        raise RedHatOwnersFileInvalidContentsError(
+            f"OWNERS file did not have expected vendor label: '{REQUIRED_VENDOR_LABEL}'"
+        )
+
+    if owners_file.get_vendor(contents) != REQUIRED_VENDOR_NAME:
+        raise RedHatOwnersFileInvalidContentsError(
+            f"OWNERS file did not have expected vendor name: '{REQUIRED_VENDOR_NAME}'"
+        )
+
+
+def main():
+    """The CLI entrypoint. Returns an exit code."""
```

Comment:
Will do, but I would argue that I don't consider them overly important. Incrementing exit codes in this (or a similar) way is an old habit, for me, originating in the time when some programs would throw nondescript error messages along with very meaningful error codes.

I generally avoid making the exit codes part of the API (by strongly defining them in documentation), but still tend to increment them because "some developer" may go back to the source and see "ah, it exited 20 which means this".

All that to say - yes, I'll happily document it here, but it's an old habit that doesn't have much backing considering I'm returning meaningful (I hope) error messages :laughing: